walk-in-interview/src/main/java/com/google/job/data/Job.java

```diff
@@ -23,4 +151,110 @@ public Job(String jobTitle) throws IllegalArgumentException {
     public String getJobTitle() {
         return jobTitle;
     }
+
+    /**
+     * Gets the location details of the job post.
+     *
+     * @return Location details of the post.
+     */
+    public Location getJobLocation() {
+        return jobLocation;
+    }
+
+    /**
+     * Gets the job description of that post. Job description cannot be empty.
+     *
+     * @return Job description of the post.
+     */
+    public String getJobDescription() {
+        return jobDescription;
+    }
+
+    /**
+     * Gets the payment details of the job post.
+     *
+     * @return Payment details of the post.
+     */
+    public JobPayment getJobPay() {
+        return jobPay;
+    }
+
+    /**
+     * Gets a list of requirements of the job post.
+     *
+     * @return Requirement list.
+     */
+    public List<String> getRequirements() {
+        return requirements;
+    }
+
+    /**
+     * Gets the date when the job post will expire.
+     *
+     * @return Expiry date of the post.
+     */
+    public long getPostExpiry() {
+        return postExpiry;
+    }
+
+    /**
+     * Gets the duration of the job itself.
+     *
+     * @return Duration of the job.
+     */
+    public JobDuration getJobDuration() {
+        return jobDuration;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        Job job = (Job) o;
+        return postExpiry == job.postExpiry &&
+                jobId.equals(job.jobId) &&
+                jobStatus == job.jobStatus &&
+                jobTitle.equals(job.jobTitle) &&
+                jobLocation.equals(job.jobLocation) &&
+                jobDescription.equals(job.jobDescription) &&
+                jobPay.equals(job.jobPay) &&
+                requirements.equals(job.requirements) &&
+                jobDuration.equals(job.jobDuration);
+    }
+
+    @Override
+    public int hashCode() {
```

Comment:
same comment about volatile hashcode and concurrency issue

Reply:
Fixed